### ISSUE TITLE
Trouble with AutoModelForSequenceClassification + Lora + Deepspeed_zero3

### ISSUE THREAD

**Issue by liuyu666-thu** [huggingface/transformers#25719]
Need to finetune a seq-classifier based on Llama-70B using Lora, but get stuck when integrating deepspeed_zero3.

Deps version:
- transformers              4.31.0
- peft                             0.5.0
- deepspeed                 0.10.1

My code is like:
1. initialize training args: `train_args = transformers.TrainingArguments(..., deepspeed=ds_config.json)`
2. create model: `model = LlamaForSequenceClassification.from_pretrained(...)`
3. wrap it with peft: `model = get_peft_model(model, lora_config)`
4. enable gradient checkpointing: `model.gradient_checkpointing_enable()`
5. train it with HF-trainer: `trainer = Trainer(..., args=train_args)`
6. launch the training using `torchrun xxx.py`

The error is like:
`{'id': 547, 'status': 'NOT_AVAILABLE', 'numel': 0, 'ds_numel': 0, 'shape': (0,), 'ds_shape': (0,), 'requires_grad': True, 'grad_shape': None, 'persist': True, 'active_sub_modules': {1482}, 'ds_tensor.shape': torch.Size([0])}`

Seems like the classification layer is sharded and something incompetible happens. 

Can anyone show the standard way to use deepspeed under hf-trainer in this scene?

**Comment by ArthurZucker** [comment]
Pretty sure you should be asking this on [the forum](https://discuss.huggingface.co/), include a full reproducer with a full traceback.

**Comment by liuyu666-thu** [comment]
> Pretty sure you should be asking this on [the forum](https://discuss.huggingface.co/), include a full reproducer with a full traceback.

Sure. I'll post one there.

**Comment by yashkens** [comment]
@liuyu666-thu Hi, did you manage to resolve this issue? I am dealing with the same problem in the same settings
